### DISCUSSION TITLE
Is there a way to disable code splitting or set max bundles to 1?

### DISCUSSION THREAD

**Discussion by wade-wojcak** [parcel-bundler/parcel#7188] [Q&A]
If I have a scenario where I need to deliver static code to another external team, and they are enforcing a requirement of only having a single js and css file, is there an easy way to enable this as part of Parcel's configuration?

**Comment by leaumar** [comment]
Wish I knew too. I'd like my webextension build to keep outputting just 1 html+js+css file for its components. The css and js are often split into multiple files for some reason. I've tried to suppress it as outlined [in the docs](https://parceljs.org/features/code-splitting/#configuration) but this config seems to have no effect. All these hashed (which I could suppress with a namer if files weren't being split) and split files are making debugging and packaging messy, to be honest. Add to this that you can't redundantly make your css and js files entrypoints (so they keep their hashless names) if your html also has to import them, and parcel ends up dictating project file structure and cli complexity.

**Reply by aloayzab88** [reply]
Did you find a way to disable code spliting and chunking?

**Reply by leaumar** [reply]
No, never found one. I actually quit my projects in the end because I was tired of how nodejs projects are 10% developing and 90% trying to make tools work. I ported my stuff to simple userscripts and just live with that now. Our projects at work have been migrated to vite where a frontend exists (vite requires html entrypoints), and we're hoping to replace parcel in the ts-only projects with bun at some point.

**Reply by aloayzab88** [reply]
I fell your pain, I'm trying to move from gulp to a modern tool, just found parcel but the issue with code spliting and chunking got me frustrated.
Btw vite has the option to avoid code spliting/chunking?

**Reply by leaumar** [reply]
I'm not sure, it didn't matter for the work projects.

**Reply by fregante** [reply]
As far as I know only rollup does, but it's a pain to setup. Vite actually does not bundle files during development node by default if I remember correctly, via rollup, but in the "production" version there's no way to do that because it uses esbuild.

**Reply by aloayzab88** [reply]
> As far as I know only rollup does, but it's a pain to setup. Vite actually does not bundle files during development node by default if I remember correctly, via rollup, but in the "production" version there's no way to do that because it uses esbuild.

So there is no way to disable code split on vite build too? Maybe I will look into webpack or stick to gulp 🥲

**Reply by mark-monteiro** [reply]
Adding this to my `package.json`, as specified in the [docs](https://parceljs.org/features/code-splitting/#configuration), worked for me

```json
  "@parcel/bundler-default": {
    "maxParallelRequests": 1
  }
```

**Comment by fregante** [comment]
Parcel appears to have added this ability via plugin, but it's not documented anywhere. Check:

- https://github.com/parcel-bundler/parcel/pull/9489

**Reply by devongovett** [reply]
I think that does the opposite: it produces one output bundle for each input file (basically not bundling).

**Reply by fregante** [reply]
Uh, I’ve been reading this whole discussion wrong for months 😭
